yt_dlp/extractor/ruv.py

```diff
@@ -99,3 +101,88 @@ def _real_extract(self, url):
             'timestamp': timestamp,
             'formats': formats,
         }
+
+
+class RuvSpilaIE(InfoExtractor):
+    IE_NAME = 'ruv.is:spila'
+    _VALID_URL = r'https?://(?:www\.)?ruv\.is/(?:(?:sjon|ut)varp|(?:krakka|ung)ruv)/spila/.+/(?P<series_id>[0-9]+)/(?P<id>[a-z0-9]+)'
+    _TESTS = [{
+        'url': 'https://www.ruv.is/sjonvarp/spila/ithrottir/30657/9jcnd4',
+        'info_dict': {
+            'id': '9jcnd4',
+            'ext': 'mp4',
+            'title': '01.02.2022',
+            'chapters': 'count:4',
+            'timestamp': 1643743500,
+            'upload_date': '20220201',
+            'thumbnail': 'https://d38kdhuogyllre.cloudfront.net/fit-in/1960x/filters:quality(65)/hd_posters/94boog-iti3jg.jpg',
+            'description': 'Íþróttafréttir.',
+            'age_limit': 0,
+        },
+    }, {
+        'url': 'https://www.ruv.is/utvarp/spila/i-ljosi-sogunnar/23795/7hqkre',
+        'info_dict': {
+            'id': '7hqkre',
+            'ext': 'mp3',
+            'thumbnail': 'https://d38kdhuogyllre.cloudfront.net/fit-in/1960x/filters:quality(65)/hd_posters/7hqkre-7uepao.jpg',
+            'description': 'md5:8d7046549daff35e9a3190dc9901a120',
+            'chapters': [],
+            'upload_date': '20220204',
+            'timestamp': 1643965500,
+            'title': 'Nellie Bly II',
+            'age_limit': 0,
+        },
+    }, {
+        'url': 'https://www.ruv.is/ungruv/spila/ungruv/28046/8beuph',
+        'only_matching': True
+    }, {
+        'url': 'https://www.ruv.is/krakkaruv/spila/krakkafrettir/30712/9jbgb0',
+        'only_matching': True
+    }]
+
+    def _real_extract(self, url):
+        display_id, series_id = self._match_valid_url(url).group('id', 'series_id')
+        program = self._download_json(
+            'https://www.ruv.is/gql/', display_id, query={'query': '''{
+                Program(id: %s){
+                    title image description short_description
+                    episodes(id: {value: "%s"}) {
+                        rating title duration file image firstrun description
+                        clips {
+                            time text
+                        }
+                        subtitles {
+                            name value
+                        }
+                    }
+                }
+            }''' % (series_id, display_id)})['data']['Program']
+        episode = program['episodes'][0]
+
+        subs = {}
+        [subs.setdefault(trk['name'], []).append({'url': trk['value'], 'ext': 'vtt'})
+         for trk in episode.get('subtitles') if trk.get('name') and trk.get('value')]
```

Comment:
```suggestion
        for trk in episode.get('subtitles'):
            if trk.get('name') and trk.get('value'):
                subs.setdefault(trk['name'], []).append({'url': trk['value'], 'ext': 'vtt'})
```
why would u use list comprehension like that !?